sql: TestLogic/5node-distsql-disk/distsql_distinct_on is flaky

Locally, I find that it sometimes fails with the following error message:

```
-- FAIL: TestLogic/5node-distsql-disk (0.00s)
        --- FAIL: TestLogic/5node-distsql-disk/distsql_distinct_on (0.57s)
            logic_test.go:2066:
                testdata/logic_test/distsql_distinct_on:56:
                expected success, but found
                pq: while adding target n1,s1: snapshot failed: (n1,s1):?: remote declined snapshot: store busy applying snapshots and/or removing replicas
```

--- a/pkg/storage/replica_command.go
+++ b/pkg/storage/replica_command.go
@@ -2385,7 +2385,8 @@ func TestingRelocateRange(
 
 	canRetry := func(err error) bool {
 		whitelist := []string{
-			"snapshot intersects existing range",
+			snapshotApplySemBusyMsg,
+			IntersectingSnapshotMsg,
 		}
 		for _, substr := range whitelist {
 			if strings.Contains(err.Error(), substr) {

